### PR TITLE
[web] Run engine tests on Safari locally by launching safari installed on MacOS

### PR DESCRIPTION
**Pull request by nturgut**
Launching safari browser on the operating system (MacOS).

Current status:
+261 methods passing  -46 methods failed.

how to run:

felt test --browser=safari

or

felt test --browser=safari --safari-version=system

soon I'll also add felt test --browser=safari --safari-version=latest for downloading and running tech preview version once we have LUCI setup.

fixes: https://github.com/flutter/flutter/issues/47231

### CODE REVIEW

File: lib/web_ui/dev/safari_installation.dart [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+// TODO(nurhan): user latest version to download and install the latest
+// technology preview.
+Future<BrowserInstallation> getOrInstallSafari(
+  String requestedVersion, {
+  StringSink infoLog,
+}) async {
+
+  // These tests are aimed to run only on MacOs machines local or on LUCI.
+  if (!io.Platform.isMacOS) {
+    throw UnimplementedError();
```
Comment (yjbanov) [inline lib/web_ui/dev/safari_installation.dart]: Let's add a message to the exception. Something like "Safari on ${io.Platform.operatingSystem} is not supported. Safari is only supported on macOS."

File: lib/web_ui/dev/safari_installation.dart [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  }
+
+  infoLog ??= io.stdout;
+
+  if (requestedVersion == 'system') {
+    // Since Safari is included in MacOS, always assume there will be one on the
+    // system.
+    infoLog.writeln('Using the system version is already installed.');
```
Comment (yjbanov) [inline lib/web_ui/dev/safari_installation.dart]: nit: remove "is"

File: lib/web_ui/dev/safari_installation.dart [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  SafariArgParser._();
+
+  @override
+  void populateOptions(ArgParser argParser) {
+    argParser
+      ..addOption(
+        'safari-version',
+        defaultsTo: 'system',
+        help: 'The Safari version to use while running tests. The Safari '
+            'browswer installed on the system is used as the only option now.'
```
[the remark this answers is not in the document]
Comment (nturgut) [reply lib/web_ui/dev/safari_installation.dart]: done.

File: lib/web_ui/dev/safari_installation.dart (hunk 1 above)
Comment (nturgut) [reply lib/web_ui/dev/safari_installation.dart]: done.

File: lib/web_ui/dev/safari_installation.dart (hunk 2 above)
Comment (nturgut) [reply lib/web_ui/dev/safari_installation.dart]: I added a `that` instead

**Pull request merged**